### ISSUE TITLE
Enhancement : a contains() that support regex

### ISSUE THREAD

**Issue by ettorerizza** [OpenRefine/OpenRefine#1554]
Issue #1550 shows that a [`contains()`](https://github.com/OpenRefine/OpenRefine/blob/master/main/src/com/google/refine/expr/functions/strings/Contains.java) function that accepts Regex might be useful. Now that @jackyq2015 has coded the [`find()`](https://github.com/OpenRefine/OpenRefine/blob/9ab897cb72111de58956721e46fb28f26f121f3b/main/src/com/google/refine/expr/functions/strings/Find.java) function, looks like even a complete beginner would be able to draw on his code to enrich `contains()`. Could it make a "good first issue" ?

**Comment by gobertm** [comment]
Hi , 
As my first issue, I'd like to take this one. 
I already tried to implement it, this looks like this : 
![image](https://user-images.githubusercontent.com/20517186/38038623-b362a6a8-32ab-11e8-8fb8-94d65ebaabe5.png)

Before rushing into (my also first) PR I'll check the whole process before doing so (test code, documentation? ) 
And is it what you requested ? 
It will work like the `partition` method second argument, meaning that `contains(string s, string sub)` will become `contains(string s, string or regex sub)`

Thank!
Maxime

**Comment by ostephens** [comment]
Hi Maxime - sounds great!

What I find useful when looking at adding functionality like this is to have a list of test cases and what the expected outcome would be. This is useful in terms of communicating how the function will work, and also gives you a set of test statements you can include in your unit tests for the functionality.

In this particular case where you are changing an existing function please make sure (and add test cases if they don't already exist) that the function continues to work as expected when used as currently intended.

**Comment by gobertm** [comment]
Hi,

(had a few days off).
I created PR #1558 .

I can not find any particular test on this function.
I tried creating mine, with simple line such as :
` Assert.assertEquals(invoke("contains", "it contains", "con"),true);`
`Assert.assertEquals(invoke("contains", "123contains", "\\d{3}"),true);`

but it seems I have to include a special evaluation function (something with Evaluable object) in order to interpret the regular expression.  I do not fully get this mechanism yet. So I created the PR as is. And look forward your input.

Regards,
Maxime
